rosbot_utils/rosbot_utils/flash-firmware.py

```diff
@@ -34,45 +35,50 @@ def __init__(self, sys_arch, binary_file):
 
         print(f"System architecture: {self.sys_arch}")
 
-        if self.sys_arch.stdout == b"armv7l\n":
+        if self.sys_arch == "armv7l":
             # Setups ThinkerBoard pins
             print("Device: ThinkerBoard\n")
             self.port = "/dev/ttyS1"
-            boot0_pin_no = 164
-            reset_pin_no = 184
+            self.boot0_pin_no = 164
+            self.reset_pin_no = 184
 
-        elif self.sys_arch.stdout == b"x86_64\n":
+        elif self.sys_arch == "x86_64":
             # Setups UpBoard pins
             print("Device: UpBoard\n")
             self.port = "/dev/ttyS4"
-            boot0_pin_no = 17
-            reset_pin_no = 18
+            self.boot0_pin_no = 17
+            self.reset_pin_no = 18
 
-        elif self.sys_arch.stdout == b"aarch64\n":
+        elif self.sys_arch == "aarch64":
             # Setups RPi pins
             print("Device: RPi\n")
             self.port = "/dev/ttyAMA0"
-            boot0_pin_no = 17
-            reset_pin_no = 18
+            self.boot0_pin_no = 17
+            self.reset_pin_no = 18
 
         else:
             print("Unknown device...")
 
-        self.boot0_pin = GPIO(boot0_pin_no, "out")
-        self.reset_pin = GPIO(reset_pin_no, "out")
+        chip = gpiod.Chip("/dev/gpiochip0")
+        self.boot0_pin = chip.request_lines(
```

Comment:
The `self.boo0_pin` looks like the port so I suggest:
 ```suggestion
        self.boot0_port = chip.request_lines(
```